server.js

```diff
@@ -14,10 +15,60 @@ const welcomeMessage = {
 //This array is our "data store".
 //We will start with one message in the array.
 //Note: messages will be lost when Glitch restarts our server.
-const messages = [welcomeMessage];
+let messages = require("./messages.json");
 
-app.get("/", function (request, response) {
-  response.sendFile(__dirname + "/index.html");
+app.get("/", (req, res) => {
+  res.sendFile(__dirname + "/index.html");
+});
+
+app.get("/messages", (req, res) => {
+  res.send(messages);
+});
+
+app.get("/messages/search", (req, res) => {
+  const searchText = req.query.text ? req.query.text.toLowerCase() : null;
+  const searchFrom = req.query.from.toLowerCase();
+  const searchMsgs = messages.filter(
+    (message) =>
+      message.text.toLowerCase().includes(searchText) ||
+      message.from.toLowerCase().includes(searchFrom)
+  );
+  res.send(searchMsgs);
+});
+
+app.get("/messages/latest", (req, res) => {
+  const latestMsgs = messages.slice(messages.length - 2, messages.length - 1);
+  res.send(latestMsgs);
+});
+
+app.get("/messages/:id", (req, res) => {
+  const inputId = req.params.id;
+  if (inputId) {
+    const welcome = messages.filter((message) => message.id == inputId);
+    res.send(inputId);
```

Comment:
This works well. It would be better to use find() rather than filter when getting by a unique ID, as this should only return 1 (or zero) results, as opposed to your search route which could return multiple results. You could also return a 404 response status if there is no result.